internal/wazeroir/operations.go

```diff
@@ -1663,52 +1676,39 @@ func NewOperationI32WrapFromI64() UnionOperation {
 //	wasm.OpcodeI32TruncSatF64SName wasm.OpcodeI32TruncSatF64UName wasm.OpcodeI64TruncSatF32SName
 //	wasm.OpcodeI64TruncSatF32UName wasm.OpcodeI64TruncSatF64SName wasm.OpcodeI64TruncSatF64UName
 //
-// See [1] and [2] for when we encounter undefined behavior in the WebAssembly specification if OperationITruncFromF.NonTrapping == false.
+// See [1] and [2] for when we encounter undefined behavior in the WebAssembly specification if NewOperationITruncFromF.NonTrapping == false.
 // To summarize, if the source float value is NaN or doesn't fit in the destination range of integers (incl. +=Inf),
 // then the runtime behavior is undefined. In wazero, the engines are expected to exit the execution in these undefined cases with
 // wasmruntime.ErrRuntimeInvalidConversionToInteger error.
 //
 // [1] https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#-hrefop-trunc-umathrmtruncmathsfu_m-n-z for unsigned integers.
 // [2] https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#-hrefop-trunc-smathrmtruncmathsfs_m-n-z for signed integers.
-type OperationITruncFromF struct {
-	InputType  Float
-	OutputType SignedInt
-	// NonTrapping true if this conversion is "nontrapping" in the sense of the
-	// https://github.com/WebAssembly/spec/blob/ce4b6c4d47eb06098cc7ab2e81f24748da822f20/proposals/nontrapping-float-to-int-conversion/Overview.md
-	NonTrapping bool
-}
-
-// String implements fmt.Stringer.
-func (o OperationITruncFromF) String() string {
-	return fmt.Sprintf("%s.%s.%s (non_trapping=%v)", o.OutputType, o.Kind(), o.InputType, o.NonTrapping)
-}
-
-// Kind implements Operation.Kind.
-func (OperationITruncFromF) Kind() OperationKind {
-	return OperationKindITruncFromF
+//
+// nonTrapping true if this conversion is "nontrapping" in the sense of the
+// https://github.com/WebAssembly/spec/blob/ce4b6c4d47eb06098cc7ab2e81f24748da822f20/proposals/nontrapping-float-to-int-conversion/Overview.md
+func NewOperationITruncFromF(inputType Float, outputType SignedInt, nonTrapping bool) UnionOperation {
+	return UnionOperation{
+		OpKind: OperationKindITruncFromF,
+		B1:     byte(inputType),
+		B2:     byte(outputType),
+		B3:     nonTrapping,
+	}
 }
 
-// OperationFConvertFromI implements Operation.
+// NewOperationFConvertFromI is a constructor for UnionOperation with Kind OperationKindFConvertFromI.
```

Comment:
`Kind` is redundant I guess? (same goes for all the comments.

```suggestion
// NewOperationFConvertFromI is a constructor for UnionOperation with OperationKindFConvertFromI.
```

Reply:
oh maybe it reads that way, the meaning was "the `Kind` field is `OperationKind<blah>`. I can remove it if it feels redundant.